### ISSUE TITLE
Permission error for drop ListenerService_1

### ISSUE THREAD

**Issue by rashkat** [dyatchenko/ServiceBrokerListener#56]
The errors I received when running the program include instruction telling me to run this script to grant the needed permissions:
 
GRANT CREATE PROCEDURE TO [MyUser];
GRANT CREATE SERVICE TO [MyUser];
GRANT CREATE QUEUE  TO [MyUser];
GRANT REFERENCES ON CONTRACT::[DEFAULT] TO [MyUser];
GRANT SUBSCRIBE QUERY NOTIFICATIONS TO [MyUser];
GRANT CONTROL ON SCHEMA::[dbo] TO [MyUser];
 
 
This script has been run and the permissions have been granted but I’m still getting an error when I run the program:
 
System.Data.SqlClient.SqlException
  HResult=0x80131904
  Message=Cannot drop the service 'ListenerService_1', because it does not exist or you do not have permission.

It would be great if you can help on this.

Regards.

**Comment by sourceship13** [comment]
I also started receiving the same error yesterday. 

```Failed to invoke hub method GetDeviceJson.
System.Data.SqlClient.SqlException (0x80131904): Cannot drop the service ListenerService_1, because it does not exist or you do not have permission.```

I'm not sure the cause, things had been working fine for me quite some time now. If anyone has seen this issue it would help us out a lot. 

Thanks,

**Comment by sourceship13** [comment]
I'm not sure if you are using Asp.Net Core, but if you are it was recently updated to version 2.1. 
It supports native SqlDependency, You may want to give that a try if it applies to you.

**Comment by dyatchenko** [comment]
HI @rashkat,

What SQL Server version do you use? Did you try running your code with admin permissions? 

Thanks,
Dmitrii

**Comment by rashkat** [comment]
Thanks for responding. It works well if I use an admin account, which I am using on my development environment, however, I don't want to deploy on production with admin rights. I will check for SQLDependency.

**Comment by dyatchenko** [comment]
The issue was fixed with #56 pr.

**Comment by jdbennettatek** [comment]
I am using this fork https://github.com/shuebener/SqlServiceBrokerListener as a NuGet package (V2.0).  Yesterday I created a new db account for the application, re-ran the GRANT permissions for the account and restarted the service.  It was fine.   As of today - each time I deploy or restart the service, I am getting a start-up exception:  Cannot drop the service 'ListenerService_1', because it does not exist or you do not have permission.  The fork I am using has this commit - so I am a bit confused as to the suddeness of this.  Is there something I should run to / change?
